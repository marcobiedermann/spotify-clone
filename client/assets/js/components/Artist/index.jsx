import PropTypes from 'prop-types';
import React from 'react';

const Artist = ({ images, name }) => (
  <figure>
    {images[0] && (
      <img src={images[0].url} alt={name} width={images[0].width} height={images[0].height} />
    )}
    <figcaption>
      <h2>
        {name}
      </h2>
    </figcaption>
  </figure>
);

Artist.propTypes = {
  images: PropTypes.arrayOf(PropTypes.shape({
    height: PropTypes.number,
    url: PropTypes.string,
    width: PropTypes.number,
  })),
  name: PropTypes.string,
};

Artist.defaultProps = {
  images: [],
  name: '',
};

export default Artist;
