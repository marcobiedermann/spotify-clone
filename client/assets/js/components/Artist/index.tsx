import React, { FC } from 'react';
import Image, { ImageProps } from '../Image';

export interface ArtistProps {
  name: string;
  id: string;
  images: ImageProps[];
}

const Artist: FC<ArtistProps> = (props) => {
  const { images, name } = props;

  return (
    <figure>
      {images[0] && <Image {...images[0]} alt={name} />}
      <figcaption>
        <h2>{name}</h2>
      </figcaption>
    </figure>
  );
};

export default Artist;
