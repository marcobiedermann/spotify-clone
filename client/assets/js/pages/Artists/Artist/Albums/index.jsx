import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Albums from '../../../../components/Albums';

class AlbumsPage extends Component {
  componentDidMount() {
    const { accessToken, fetchArtistAlbums } = this.props;

    fetchArtistAlbums(accessToken, '20JZFwl6HVl6yg8a4H3ZqK');
  }

  render() {
    const { albums } = this.props;

    return (
      <Albums albums={albums} />
    );
  }
}

AlbumsPage.propTypes = {
  accessToken: PropTypes.string,
  albums: PropTypes.arrayOf(),
  fetchArtistAlbums: PropTypes.func,
};

AlbumsPage.defaultProps = {
  accessToken: '',
  albums: [],
  fetchArtistAlbums: () => {},
};

export default AlbumsPage;
