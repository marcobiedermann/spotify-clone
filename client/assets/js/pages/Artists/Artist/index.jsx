import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  Link,
  Route,
  Switch,
} from 'react-router-dom';
import ArtistAlbumsPageContainer from '../../../containers/ArtistAlbumsPage';
import ArtistRelatedArtistsPageContainer from '../../../containers/ArtistRelatedArtistsPage';
import ArtistTopTracksPageContainer from '../../../containers/ArtistTopTracksPage';
import Artist from '../../../components/Artist';

class ArtistPage extends Component {
  componentDidMount() {
    const { accessToken, fetchArtist, match } = this.props;

    fetchArtist(accessToken, match.params.id);
  }

  render() {
    const { artist, match } = this.props;

    return (
      <Switch>
        <Route
          path={`${match.url}/albums`}
          component={ArtistAlbumsPageContainer}
        />
        <Route
          path={`${match.url}/related-artists`}
          component={ArtistRelatedArtistsPageContainer}
        />
        <Route
          path={`${match.url}/top-tracks`}
          component={ArtistTopTracksPageContainer}
        />
        <Route
          path={match.url}
          component={() => (
            <div>
              <Artist {...artist} />
              <ul>
                <li>
                  <Link to={`${match.url}/albums`}>Albums</Link>
                </li>
                <li>
                  <Link to={`${match.url}/related-artists`}>Related Artists</Link>
                </li>
                <li>
                  <Link to={`${match.url}/top-tracks`}>Top Tracks</Link>
                </li>
              </ul>
            </div>
          )}
        />
      </Switch>
    );
  }
}

ArtistPage.propTypes = {
  accessToken: PropTypes.string,
  artist: PropTypes.shape(),
  fetchArtist: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
    url: PropTypes.string,
  }),
};

ArtistPage.defaultProps = {
  accessToken: '',
  artist: null,
  fetchArtist: () => {},
  match: {
    params: {
      id: '',
    },
    url: '',
  },
};

export default ArtistPage;
