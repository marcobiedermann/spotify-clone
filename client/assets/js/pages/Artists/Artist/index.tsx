import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { Link, RouteChildrenProps, useParams } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchArtist } from '../../../actions/artists';
import Artist, { ArtistProps } from '../../../components/Artist';
import Error from '../../../components/Error';
import Loader from '../../../components/Loader';

export interface ArtistPageProps extends RouteChildrenProps {
  accessToken: string;
  artist: ArtistProps;
  error: {
    message: string;
  };
  fetchArtist: any;
  isLoading: boolean;
}

export class ArtistPage extends Component<ArtistPageProps> {
  componentDidMount() {
    const { accessToken, fetchArtist } = this.props;
    const { artistId } = useParams();

    fetchArtist(accessToken, artistId);
  }

  render() {
    const { artist, error, isLoading, match } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <>
        <Helmet>
          <title>{artist.name}</title>
        </Helmet>
        <Artist {...artist} />
        <ul>
          <li>
            <Link to={`${match.url}/albums`}>Albums</Link>
          </li>
          <li>
            <Link to={`${match.url}/related-artists`}>Related Artists</Link>
          </li>
          <li>
            <Link to={`${match.url}/top-tracks`}>Top Tracks</Link>
          </li>
        </ul>
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  artist: state.artists.artist,
  error: state.albums.error,
  isLoading: state.artists.isLoading,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchArtist,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(ArtistPage);
