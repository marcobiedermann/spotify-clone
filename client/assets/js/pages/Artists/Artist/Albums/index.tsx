import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { RouteChildrenProps } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchArtistAlbums } from '../../../../actions/artists';
import Albums, { AlbumsProps } from '../../../../components/Albums';
import Error from '../../../../components/Error';
import Loader from '../../../../components/Loader';

export interface AlbumsPageProps extends RouteChildrenProps {
  accessToken: string;
  albums: AlbumsProps;
  error: {
    message: string;
  };
  fetchArtistAlbums: any;
  isLoading: boolean;
}

export class AlbumsPage extends Component<AlbumsPageProps> {
  componentDidMount() {
    const { accessToken, fetchArtistAlbums } = this.props;

    fetchArtistAlbums(accessToken, '20JZFwl6HVl6yg8a4H3ZqK');
  }

  render() {
    const { albums, error, isLoading } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <>
        <Helmet>
          <title>Albums</title>
        </Helmet>
        <Albums {...albums} />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  albums: state.artists.albums,
  error: state.artists.error,
  isLoading: state.artists.isLoading,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchArtistAlbums,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(AlbumsPage);
