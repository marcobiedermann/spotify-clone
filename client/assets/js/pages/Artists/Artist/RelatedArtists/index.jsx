import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchArtistRelatedArtists } from '../../../../actions/artists';
import Artists from '../../../../components/Artists';
import Error from '../../../../components/Error';
import Loader from '../../../../components/Loader';

export class RelatedArtistsPage extends Component {
  static propTypes = {
    accessToken: PropTypes.string,
    artists: PropTypes.arrayOf(PropTypes.shape),
    error: PropTypes.shape({
      message: PropTypes.string,
    }),
    fetchArtistRelatedArtists: PropTypes.func,
    isLoading: PropTypes.bool,
  };

  static defaultProps = {
    accessToken: '',
    artists: [],
    error: {
      message: '',
    },
    fetchArtistRelatedArtists: () => {},
    isLoading: false,
  };

  componentDidMount() {
    const { accessToken, fetchArtistRelatedArtists } = this.props;

    fetchArtistRelatedArtists(accessToken, '20JZFwl6HVl6yg8a4H3ZqK');
  }

  render() {
    const { artists, error, isLoading } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <div>
        <Helmet>
          <title>Related Artists</title>
        </Helmet>
        <Artists artists={artists} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  artists: state.artists.artists,
  error: state.artists.error,
  isLoading: state.artists.isLoading,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchArtistRelatedArtists,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(RelatedArtistsPage);
