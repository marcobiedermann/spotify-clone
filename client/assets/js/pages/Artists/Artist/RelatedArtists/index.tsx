import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { RouteChildrenProps } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchArtistRelatedArtists } from '../../../../actions/artists';
import { ArtistProps } from '../../../../components/Artist';
import Artists from '../../../../components/Artists';
import Error from '../../../../components/Error';
import Loader from '../../../../components/Loader';

export interface RelatedArtistsPageProps extends RouteChildrenProps {
  accessToken: string;
  artists: ArtistProps[];
  error: {
    message: string;
  };
  fetchArtistRelatedArtists: any;
  isLoading: boolean;
}

export class RelatedArtistsPage extends Component<RelatedArtistsPageProps> {
  componentDidMount() {
    const { accessToken, fetchArtistRelatedArtists } = this.props;

    fetchArtistRelatedArtists(accessToken, '20JZFwl6HVl6yg8a4H3ZqK');
  }

  render() {
    const { artists, error, isLoading } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <>
        <Helmet>
          <title>Related Artists</title>
        </Helmet>
        <Artists artists={artists} />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  artists: state.artists.artists,
  error: state.artists.error,
  isLoading: state.artists.isLoading,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchArtistRelatedArtists,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(RelatedArtistsPage);
