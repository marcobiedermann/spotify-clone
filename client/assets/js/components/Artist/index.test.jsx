import React from 'react';
import renderer from 'react-test-renderer';
import Artist from '.';

it('renders correctly', () => {
  const tree = renderer
    .create(<Artist />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
