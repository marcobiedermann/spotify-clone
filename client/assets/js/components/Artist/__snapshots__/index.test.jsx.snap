// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders correctly 1`] = `
<figure>
  <img
    alt=""
    height={0}
    src=""
    width={0}
  />
  <figcaption>
    <h2>
      
    </h2>
  </figcaption>
</figure>
`;
